mloop/neuralnet.py

```diff
@@ -92,34 +87,40 @@ def __init__(self,
 
         with self.graph.as_default():
             ## Inputs
-            self.input_placeholder = tf.placeholder(tf.float32, shape=[None, self.num_params])
-            self.output_placeholder = tf.placeholder(tf.float32, shape=[None, 1])
-            self.keep_prob_placeholder = tf.placeholder_with_default(1., shape=[])
-            self.regularisation_coefficient_placeholder = tf.placeholder_with_default(0., shape=[])
+            self.input_placeholder = tf.compat.v1.placeholder(
+                tf.float32, shape=[None, self.num_params])
+            self.output_placeholder = tf.compat.v1.placeholder(
+                tf.float32, shape=[None, 1])
+            self.keep_prob_placeholder = tf.compat.v1.placeholder_with_default(
+                1., shape=[])
+            self.regularisation_coefficient_placeholder = tf.compat.v1.placeholder_with_default(
+                0., shape=[])
 
             ## Initialise the network
 
             weights = []
             biases = []
 
+            weight_initializer = tf.compat.v1.initializers.he_normal()
+            bias_initializer = tf.compat.v1.random_normal_initializer(stddev=0.5)
+
             # Input + internal nodes
             prev_layer_dim = self.num_params
-            bias_stddev=0.5
             for (i, dim) in enumerate(layer_dims):
                 weights.append(tf.Variable(
-                    tf.random_normal([prev_layer_dim, dim], stddev=1.4/np.sqrt(prev_layer_dim)),
+                    weight_initializer(shape=[prev_layer_dim, dim], dtype=tf.float32),
                     name="weight_"+str(i)))
                 biases.append(tf.Variable(
-                    tf.random_normal([dim], stddev=bias_stddev),
+                    bias_initializer(shape=[1], dtype=tf.float32),
```

Comment:
I guess 1 also works here, but any reason we want to use the same value for all nodes?

Reply:
D'oh, well spotted!